File: ClassicGuildBankData/Models/RequestItemsModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ClassicGuildBankData.Models
{
    public class RequestItemsModel
    {
        #region Properties

        public string CharacterName
        { 
            get; 
            set; 
        }

        public int Gold
        {
            get;
            set;
        }

        [Required]
        public Guid GuildId
        {
            get;
            set;
        }

        private List<RequestItemModel> _requestItemModels;
        public List<RequestItemModel> RequestItemModels
        {
            get
            {
                if (_requestItemModels == null)
                    _requestItemModels = new List<RequestItemModel>();

                return _requestItemModels;
            }
            set
            {
                _requestItemModels = value;
            }
        }

        #endregion
    }
}

File: classic-guild-bank/src/app/models/guild/requestItemsModel.ts
import { RequestItemModel } from "./requestItemModel";

export class RequestItemsModel {
    public characterName: string;
    public gold: number;
    public guildId: string;
    public requestItemModels: RequestItemModel[];

    constructor(init?: Partial<RequestItemsModel>){
        if (init) {
            Object.assign(this, init);
        }
    }
}